src/modules/launcher/PowerLauncher/MainWindow.xaml.cs

```diff
@@ -307,6 +333,10 @@ private void _launcher_KeyDown(object sender, KeyRoutedEventArgs e)
                 _viewModel.SelectPrevPageCommand.Execute(null);
                 e.Handled = true;
             }
+            else if( e.Key == VirtualKey.Back)
+            {
+                _deletePressed = true;
+            }

```

Comment:
There might also be another approach in which the user deletes the query, by pressing `Ctrl+A` and `delete`. I normally do that when the first keystroke is missed and this does not capture that case.

Reply:
@alekhyareddy28 This is why I went with a timer.  If you do Ctrl-A / Delete within 2 seconds it should fire this event.  The two second limit is essentially there to try to differentiate between issues caused because of eaten keyboard input vs the user changing there mind on what they wanted.   @jyuwono I think we need a 'time to focus' event as well though to capture scenarios where the textbox wasn't focused after showing it.  In this case the user won't be deleting anything in the 2 second window because they don't have focus.

Reply:
Sounds good. I think I missed that.